core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -402,7 +402,12 @@ private[spark] class Executor(
           // Don't forcibly exit unless the exception was inherently fatal, to avoid
           // stopping other tasks unnecessarily.
           if (Utils.isFatalError(t)) {
-            SparkUncaughtExceptionHandler.uncaughtException(t)
+            if (!isLocal) {
+              Thread.getDefaultUncaughtExceptionHandler.
```

Comment:
The executors would not remain up. It will go down and come up again. That's what SnappyUncaughExceptionHandler would do. 

You mean even for other errors we should restart the executor?